bittensor/core/async_subtensor.py

```diff
@@ -350,6 +351,39 @@ async def metagraph(
 
         return metagraph
 
+    async def get_metagraph(
+        self, netuid: int, block: Optional[int] = None
+    ) -> Optional["MetagraphInfo"]:
+        if block is not None:
+            block_hash = await self.get_block_hash(block)
+        else:
+            block_hash = None
```

Comment:
```suggestion
        block_hash = await self.get_block_hash(block)
```
We already have the logic to handle None inside here.